src/core/Table/hooks/useColumnDragAndDrop.tsx

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import {
+  HeaderGroup,
+  Hooks,
+  IdType,
+  makePropGetter,
+  TableInstance,
+  TableKeyedProps,
+  useGetLatest,
+} from 'react-table';
+
+export const useColumnDragAndDrop = <T extends Record<string, unknown>>(
+  isEnabled: boolean,
+) => (hooks: Hooks<T>) => {
+  hooks.getDragAndDropProps = [defaultGetDragAndDropProps(isEnabled)];
```

Comment:
`defaultGetDragAndDropProps` is generic. Dont you need to pass it here? Is it even needed to be generic?

Reply:
I don't know why it works without passing explicitly but this where generic type is needed 
![image](https://user-images.githubusercontent.com/36186912/158202113-5a1429b9-4b55-48fc-9793-daa49bb5d8c9.png)